WordPress/gutenberg
# Popover: add `box-sizing` reset style

Fixes #58865
Fixes #56190
Alternative to #56325

## What?

This PR applies `box-sizing` reset style to the `Popover` component itself to prevent unintended horizontal scrollbars from appearing in any context. This will prevent unintended scroll bars from appearing in the popover of the navigation block in the Site Editor.

| Before | After |
|--------|--------|
| ![image](https://github.com/WordPress/gutenberg/assets/54422211/0ef11cc2-facb-4c73-8b09-8b992da22ebf)| ![image](https://github.com/WordPress/gutenberg/assets/54422211/fcb3193c-7c12-45ef-bc77-7f806e386e84) | 

## Why?

Box sizing reset style for Popover was being applied on each editor instance, not on the component itself. However, unlike the post editor, this reset was not applied in the Site Editor.

https://github.com/WordPress/gutenberg/blob/9ca92adec8303df0094b6eebaeed714df69a356a/packages/edit-site/src/style.scss#L70-L74

I suspect that this is the cause of this issue.

## How?

Applied a box sizing reset style to the component itself and removed the reset mixin when it was no longer needed.

As a further step, we might also apply a similar approach to the `Modal` component.

https://github.com/WordPress/gutenberg/blob/31d5bd0167fb6667be6695a83bf6a94bc1701f3c/packages/edit-post/src/style.scss#L51-L53
https://github.com/WordPress/gutenberg/blob/31d5bd0167fb6667be6695a83bf6a94bc1701f3c/packages/edit-site/src/style.scss#L72-L74
https://github.com/WordPress/gutenberg/blob/31d5bd0167fb6667be6695a83bf6a94bc1701f3c/packages/edit-widgets/src/style.scss#L21-L22
## Testing Instructions

- Open the Site Editor and select the navigation block.
- Add a new link.
- The popover showing links should not display scrollbars.
